src/app/tests/suites/include/ConstraintsChecker.h

```diff
@@ -245,6 +245,20 @@ class ConstraintsChecker
         return CheckConstraintMaxValue(itemName, current.Value(), static_cast<T>(expected));
     }
 
+    template <typename T, typename U, std::enable_if_t<!std::is_enum<T>::value, int> = 0>
+    bool CheckConstraintNotValue(const char * itemName, const chip::app::DataModel::Nullable<T> & current,
+                                 const chip::app::DataModel::Nullable<U> & expected)
+    {
+        if (current == expected)
+        {
+            std::string expectedString(expected.HasValue() ? expected.Value() : "NULL");
+            std::string currentString(current.HasValue() ? current.Value() : "NULL");
+            Exit(std::string(itemName) + " expected " + expected + " got " + expectedString + " instead.");
```

Comment:
This is an odd message because by assumption on this branch current == expected.

And this is not using `currentString` at all.

And the whole point is that we did _not_ expect `expected`, but did get it.

Reply:
Hmm... Yeah, that makes sense. Updated